{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "13342e66",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "\n",
    "Below are the code snippets and results for day 2 of the practical course accompanying the BIOMEDAS lecture \"Comparative Genomics, Phylogenetics and other Sequence-based Analyses\"."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c521b065",
   "metadata": {},
   "source": [
    "# Alignment\n",
    "\n",
    "We aim at the following:\n",
    "\n",
    " * extract our novel flaviviral protein sequence\n",
    " * extract the HMM profile of the best hit\n",
    " * create alignment of new and reference flaviruses based on the pHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1cf06fc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# extract discovered flavivirus protein sequence\n",
    "# on Day 1 we found that the sequence with ID '2049' is from a flavivirus\n",
    "!grep -A 1 '>2049' ../Day1/SRR5945842_plass_clustered_rep_seq.fasta > 2049.fasta\n",
    "\n",
    "# extract Flavi pHMM from Pfam database\n",
    "# on Day 1 we found that this is the profile with ID 'Flavi_DEAD'\n",
    "!hmmfetch ../Day1/Pfam-A.hmm Flavi_DEAD > Flavi_DEAD.hmm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a7e12176",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add sequence to Pfam alignment\n",
    "#  requires downloading of Flavi_DEAD seed alignment in Fasta format saved in file Flavi_DEAD.fasta\n",
    "!hmmalign --mapali Flavi_DEAD.fasta --outformat A2M Flavi_DEAD.hmm 2049.fasta > Flavi_DEAD_2049.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fa7ff3c",
   "metadata": {},
   "source": [
    "# Phylogeny"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6b777cb8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                             _      _ _            _      _   _  _____ \n",
      "                            | |    | | |          | |    | \\ | |/ ____|\n",
      "         _ __ ___   ___   __| | ___| | |_ ___  ___| |_   |  \\| | |  __ \n",
      "        | '_ ` _ \\ / _ \\ / _` |/ _ \\ | __/ _ \\/ __| __|  | . ` | | |_ |\n",
      "        | | | | | | (_) | (_| |  __/ | ||  __/\\__ \\ |_   | |\\  | |__| |\n",
      "        |_| |_| |_|\\___/ \\__,_|\\___|_|\\__\\___||___/\\__|  |_| \\_|\\_____|\n",
      "--------------------------------------------------------------------------------\n",
      "ModelTest-NG v0.1.7 released on 17.03.2021 by The Exelixis Lab.\n",
      "Written by Diego Darriba.\n",
      "Contributors: Tomas Flouri, Alexey Kozlov, Benoit Morel, David Posada, \n",
      "              Alexandros Stamatakis.\n",
      "Latest version: https://github.com/ddarriba/modeltest\n",
      "--------------------------------------------------------------------------------\n",
      "\n",
      "Physical cores: 4\n",
      "Logical cores:  8\n",
      "Memory:         15.3GB\n",
      "Extensions:     AVX\n",
      "\n",
      "WARNING: MSA has not enough sites to infer reliable results\n",
      "Creating new checkpoint file: Flavi_DEAD_2049_Gblocks.fasta.ckp\n",
      "--------------------------------------------------------------------------------\n",
      "ModelTest-NG v0.1.7\n",
      "\n",
      "Input data:\n",
      "  MSA:        Flavi_DEAD_2049_Gblocks.fasta\n",
      "  Tree:       Maximum parsimony\n",
      "    file:           -\n",
      "  #taxa:            20\n",
      "  #sites:           74\n",
      "  #patterns:        66\n",
      "  Max. thread mem:  3 MB\n",
      "\n",
      "Output:\n",
      "  Log:           Flavi_DEAD_2049_Gblocks.fasta.log\n",
      "  Starting tree: Flavi_DEAD_2049_Gblocks.fasta.tree\n",
      "  Results:       Flavi_DEAD_2049_Gblocks.fasta.out\n",
      "\n",
      "Selection options:\n",
      "  # protein matrices: 19\n",
      "  # protein models:   152\n",
      "  include model parameters:\n",
      "    Uniform:         true\n",
      "    p-inv (+I):      true\n",
      "    gamma (+G):      true\n",
      "    both (+I+G):     true\n",
      "    free rates (+R): false\n",
      "    fixed freqs:     true\n",
      "    estimated freqs: false\n",
      "    #categories:     4\n",
      "  gamma rates mode:   mean\n",
      "  asc bias:           none\n",
      "  epsilon (opt):      0.01\n",
      "  epsilon (par):      0.05\n",
      "  keep branches:      false\n",
      "\n",
      "Additional options:\n",
      "  verbosity:        very low\n",
      "  threads:          1/4\n",
      "  RNG seed:         12345\n",
      "  subtree repeats:  enabled\n",
      "--------------------------------------------------------------------------------\n",
      "modeltest-ng was called as follows: \n",
      ">> ./modeltest-ng -i Flavi_DEAD_2049_Gblocks.fasta -d aa \n",
      "\n",
      "\n",
      "modeltest-ng: Warning: You are using one single thread out of 4 physical cores.\n",
      "modeltest-ng:          You can set the number of threads with -p argument.\n",
      "modeltest-ng: Try 'modeltest-ng --help' or 'modeltest-ng --usage' for more information\n",
      "\n",
      "Partition 1/1\n",
      "\n",
      "    1/152  DAYHOFF        0h:00:00   0h:00:00           -1645.9594       -       -\n",
      "    2/152  DAYHOFF+I      0h:00:00   0h:00:00           -1585.1540       -  0.2293\n",
      "    3/152  DAYHOFF+G4     0h:00:00   0h:00:00           -1559.5234  0.6507       -\n",
      "    4/152  DAYHOFF+I+G4   0h:00:00   0h:00:00           -1556.4477  1.4925  0.2021\n",
      "    5/152  DAYHOFF+F      0h:00:00   0h:00:00           -1643.9993       -       -\n",
      "    6/152  DAYHOFF+I+F    0h:00:00   0h:00:00           -1588.2536       -  0.2294\n",
      "    7/152  DAYHOFF+G4+F   0h:00:00   0h:00:00           -1565.2701  0.6774       -\n",
      "    8/152  DAYHOFF+I+G4+F 0h:00:00   0h:00:00           -1562.2951  1.5664  0.1999\n",
      "    9/152  LG             0h:00:00   0h:00:00           -1608.8089       -       -\n",
      "   10/152  LG+I           0h:00:00   0h:00:00           -1560.7532       -  0.2244\n",
      "   11/152  LG+G4          0h:00:00   0h:00:00           -1528.9082  0.6782       -\n",
      "   12/152  LG+I+G4        0h:00:00   0h:00:00           -1527.3034  1.1341  0.1633\n",
      "   13/152  LG+F           0h:00:00   0h:00:00           -1609.3952       -       -\n",
      "   14/152  LG+I+F         0h:00:00   0h:00:00           -1565.8119       -  0.2239\n",
      "   15/152  LG+G4+F        0h:00:00   0h:00:00           -1535.4782  0.6804       -\n",
      "   16/152  LG+I+G4+F      0h:00:01   0h:00:01           -1534.1620  1.2111  0.1593\n",
      "   17/152  DCMUT          0h:00:00   0h:00:01           -1645.5825       -       -\n",
      "   18/152  DCMUT+I        0h:00:00   0h:00:01           -1584.8963       -  0.2293\n",
      "   19/152  DCMUT+G4       0h:00:00   0h:00:01           -1559.4912  0.6528       -\n",
      "   20/152  DCMUT+I+G4     0h:00:00   0h:00:01           -1556.4084  1.5008  0.2023\n",
      "   21/152  DCMUT+F        0h:00:00   0h:00:01           -1643.6103       -       -\n",
      "   22/152  DCMUT+I+F      0h:00:00   0h:00:01           -1587.9758       -  0.2294\n",
      "   23/152  DCMUT+G4+F     0h:00:00   0h:00:01           -1565.1887  0.6792       -\n",
      "   24/152  DCMUT+I+G4+F   0h:00:00   0h:00:01           -1562.2110  1.5648  0.1997\n",
      "   25/152  JTT            0h:00:00   0h:00:01           -1638.9995       -       -\n",
      "   26/152  JTT+I          0h:00:00   0h:00:01           -1580.9308       -  0.2297\n",
      "   27/152  JTT+G4         0h:00:00   0h:00:01           -1558.0931  0.6537       -\n",
      "   28/152  JTT+I+G4       0h:00:00   0h:00:01           -1554.9127  1.4947  0.2020\n",
      "   29/152  JTT+F          0h:00:00   0h:00:01           -1650.0953       -       -\n",
      "   30/152  JTT+I+F        0h:00:00   0h:00:01           -1595.9060       -  0.2299\n",
      "   31/152  JTT+G4+F       0h:00:00   0h:00:01           -1571.7378  0.6517       -\n",
      "   32/152  JTT+I+G4+F     0h:00:00   0h:00:01           -1568.9389  1.4407  0.1952\n",
      "   33/152  MTREV          0h:00:00   0h:00:01           -1750.6539       -       -\n",
      "   34/152  MTREV+I        0h:00:00   0h:00:01           -1701.0742       -  0.2143\n",
      "   35/152  MTREV+G4       0h:00:00   0h:00:01           -1630.2130  0.5330       -\n",
      "   36/152  MTREV+I+G4     0h:00:00   0h:00:01           -1630.2293  0.7566  0.1044\n",
      "   37/152  MTREV+F        0h:00:00   0h:00:01           -1711.2295       -       -\n",
      "   38/152  MTREV+I+F      0h:00:00   0h:00:01           -1661.3767       -  0.2177\n",
      "   39/152  MTREV+G4+F     0h:00:00   0h:00:01           -1603.9705  0.5354       -\n",
      "   40/152  MTREV+I+G4+F   0h:00:01   0h:00:02           -1603.6137  1.0000  0.1544\n",
      "   41/152  WAG            0h:00:00   0h:00:02           -1610.6300       -       -\n",
      "   42/152  WAG+I          0h:00:00   0h:00:02           -1561.6119       -  0.2271\n",
      "   43/152  WAG+G4         0h:00:00   0h:00:02           -1541.6196  0.7391       -\n",
      "   44/152  WAG+I+G4       0h:00:00   0h:00:02           -1539.1621  1.6444  0.1946\n",
      "   45/152  WAG+F          0h:00:00   0h:00:02           -1609.9328       -       -\n",
      "   46/152  WAG+I+F        0h:00:00   0h:00:02           -1565.7038       -  0.2268\n",
      "   47/152  WAG+G4+F       0h:00:00   0h:00:02           -1547.1757  0.7397       -\n",
      "   48/152  WAG+I+G4+F     0h:00:00   0h:00:02           -1544.8984  1.6744  0.1882\n",
      "   49/152  RTREV          0h:00:00   0h:00:02           -1619.2277       -       -\n",
      "   50/152  RTREV+I        0h:00:00   0h:00:02           -1573.9242       -  0.2209\n",
      "   51/152  RTREV+G4       0h:00:00   0h:00:02           -1541.8393  0.6780       -\n",
      "   52/152  RTREV+I+G4     0h:00:00   0h:00:02           -1540.4640  1.0892  0.1487\n",
      "   53/152  RTREV+F        0h:00:00   0h:00:02           -1616.4231       -       -\n",
      "   54/152  RTREV+I+F      0h:00:00   0h:00:02           -1574.2998       -  0.2211\n",
      "   55/152  RTREV+G4+F     0h:00:00   0h:00:02           -1540.3059  0.6827       -\n",
      "   56/152  RTREV+I+G4+F   0h:00:00   0h:00:02           -1539.2683  1.1012  0.1404\n",
      "   57/152  CPREV          0h:00:00   0h:00:02           -1638.5596       -       -\n",
      "   58/152  CPREV+I        0h:00:00   0h:00:02           -1593.8143       -  0.2242\n",
      "   59/152  CPREV+G4       0h:00:00   0h:00:02           -1567.3302  0.6815       -\n",
      "   60/152  CPREV+I+G4     0h:00:01   0h:00:03           -1565.8250  1.3969  0.1787\n",
      "   61/152  CPREV+F        0h:00:00   0h:00:03           -1629.6712       -       -\n",
      "   62/152  CPREV+I+F      0h:00:00   0h:00:03           -1586.8181       -  0.2232\n",
      "   63/152  CPREV+G4+F     0h:00:00   0h:00:03           -1562.5155  0.7300       -\n",
      "   64/152  CPREV+I+G4+F   0h:00:00   0h:00:03           -1560.7468  1.4845  0.1760\n",
      "   65/152  VT             0h:00:00   0h:00:03           -1615.6281       -       -\n",
      "   66/152  VT+I           0h:00:00   0h:00:03           -1566.1534       -  0.2287\n",
      "   67/152  VT+G4          0h:00:00   0h:00:03           -1547.6191  0.7454       -\n",
      "   68/152  VT+I+G4        0h:00:00   0h:00:03           -1544.9913  1.6795  0.1970\n",
      "   69/152  VT+F           0h:00:00   0h:00:03           -1619.2935       -       -\n",
      "   70/152  VT+I+F         0h:00:00   0h:00:03           -1573.6081       -  0.2281\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   71/152  VT+G4+F        0h:00:00   0h:00:03           -1556.3238  0.7512       -\n",
      "   72/152  VT+I+G4+F      0h:00:00   0h:00:03           -1553.8386  1.7329  0.1925\n",
      "   73/152  BLOSUM62       0h:00:00   0h:00:03           -1613.6384       -       -\n",
      "   74/152  BLOSUM62+I     0h:00:00   0h:00:03           -1571.1713       -  0.2261\n",
      "   75/152  BLOSUM62+G4    0h:00:00   0h:00:03           -1549.0645  0.7502       -\n",
      "   76/152  BLOSUM62+I+G4  0h:00:00   0h:00:03           -1547.4543  1.4335  0.1710\n",
      "   77/152  BLOSUM62+F     0h:00:00   0h:00:03           -1614.1209       -       -\n",
      "   78/152  BLOSUM62+I+F   0h:00:00   0h:00:03           -1574.0545       -  0.2260\n",
      "   79/152  BLOSUM62+G4+F  0h:00:00   0h:00:03           -1555.5253  0.7706       -\n",
      "   80/152  BLOSUM62+I+G4+F0h:00:01   0h:00:04           -1553.8384  1.5730  0.1730\n",
      "   81/152  MTMAM          0h:00:00   0h:00:04           -1850.1459       -       -\n",
      "   82/152  MTMAM+I        0h:00:00   0h:00:04           -1792.5012       -  0.2126\n",
      "   83/152  MTMAM+G4       0h:00:00   0h:00:04           -1683.0602  0.4766       -\n",
      "   84/152  MTMAM+I+G4     0h:00:00   0h:00:04           -1682.3895  0.5592  0.0905\n",
      "   85/152  MTMAM+F        0h:00:00   0h:00:04           -1813.4825       -       -\n",
      "   86/152  MTMAM+I+F      0h:00:00   0h:00:04           -1755.6243       -  0.2167\n",
      "   87/152  MTMAM+G4+F     0h:00:00   0h:00:04           -1660.2276  0.4568       -\n",
      "   88/152  MTMAM+I+G4+F   0h:00:00   0h:00:04           -1659.6047  0.7106  0.1397\n",
      "   89/152  MTART          0h:00:00   0h:00:04           -1800.7781       -       -\n",
      "   90/152  MTART+I        0h:00:00   0h:00:04           -1756.9722       -  0.1983\n",
      "   91/152  MTART+G4       0h:00:00   0h:00:04           -1628.6896  0.4863       -\n",
      "   92/152  MTART+I+G4     0h:00:00   0h:00:04           -1628.4126  0.5086  0.0519\n",
      "   93/152  MTART+F        0h:00:00   0h:00:04           -1744.5532       -       -\n",
      "   94/152  MTART+I+F      0h:00:00   0h:00:04           -1700.1778       -  0.2006\n",
      "   95/152  MTART+G4+F     0h:00:00   0h:00:04           -1603.9331  0.4820       -\n",
      "   96/152  MTART+I+G4+F   0h:00:00   0h:00:04           -1603.7626  0.5965  0.0816\n",
      "   97/152  MTZOA          0h:00:00   0h:00:04           -1731.6840       -       -\n",
      "   98/152  MTZOA+I        0h:00:00   0h:00:04           -1688.5023       -  0.2024\n",
      "   99/152  MTZOA+G4       0h:00:00   0h:00:04           -1592.8280  0.5044       -\n",
      "  100/152  MTZOA+I+G4     0h:00:01   0h:00:05           -1592.8129  0.5633  0.0458\n",
      "  101/152  MTZOA+F        0h:00:00   0h:00:05           -1691.9537       -       -\n",
      "  102/152  MTZOA+I+F      0h:00:00   0h:00:05           -1646.5455       -  0.2081\n",
      "  103/152  MTZOA+G4+F     0h:00:00   0h:00:05           -1576.9278  0.5115       -\n",
      "  104/152  MTZOA+I+G4+F   0h:00:00   0h:00:05           -1577.0212  0.5737  0.0515\n",
      "  105/152  PMB            0h:00:00   0h:00:05           -1613.8090       -       -\n",
      "  106/152  PMB+I          0h:00:00   0h:00:05           -1572.6531       -  0.2266\n",
      "  107/152  PMB+G4         0h:00:00   0h:00:05           -1550.6708  0.7493       -\n",
      "  108/152  PMB+I+G4       0h:00:00   0h:00:05           -1549.1805  1.4205  0.1674\n",
      "  109/152  PMB+F          0h:00:00   0h:00:05           -1614.5569       -       -\n",
      "  110/152  PMB+I+F        0h:00:00   0h:00:05           -1575.6112       -  0.2263\n",
      "  111/152  PMB+G4+F       0h:00:00   0h:00:05           -1556.3364  0.7765       -\n",
      "  112/152  PMB+I+G4+F     0h:00:00   0h:00:05           -1554.8303  1.6439  0.1762\n",
      "  113/152  HIVB           0h:00:00   0h:00:05           -1710.7377       -       -\n",
      "  114/152  HIVB+I         0h:00:00   0h:00:05           -1643.0738       -  0.2280\n",
      "  115/152  HIVB+G4        0h:00:00   0h:00:05           -1603.2341  0.5647       -\n",
      "  116/152  HIVB+I+G4      0h:00:00   0h:00:05           -1600.6037  1.1847  0.1970\n",
      "  117/152  HIVB+F         0h:00:00   0h:00:05           -1717.6967       -       -\n",
      "  118/152  HIVB+I+F       0h:00:00   0h:00:05           -1654.5116       -  0.2280\n",
      "  119/152  HIVB+G4+F      0h:00:01   0h:00:06           -1608.5779  0.5225       -\n",
      "  120/152  HIVB+I+G4+F    0h:00:00   0h:00:06           -1606.6231  1.1251  0.1883\n",
      "  121/152  HIVW           0h:00:00   0h:00:06           -1783.0820       -       -\n",
      "  122/152  HIVW+I         0h:00:00   0h:00:06           -1703.5962       -  0.2281\n",
      "  123/152  HIVW+G4        0h:00:00   0h:00:06           -1668.1429  0.5647       -\n",
      "  124/152  HIVW+I+G4      0h:00:00   0h:00:06           -1664.1088  1.2923  0.2114\n",
      "  125/152  HIVW+F         0h:00:00   0h:00:06           -1770.3384       -       -\n",
      "  126/152  HIVW+I+F       0h:00:00   0h:00:06           -1698.4752       -  0.2291\n",
      "  127/152  HIVW+G4+F      0h:00:00   0h:00:06           -1659.6868  0.5293       -\n",
      "  128/152  HIVW+I+G4+F    0h:00:00   0h:00:06           -1657.0980  1.3200  0.2140\n",
      "  129/152  JTT-DCMUT      0h:00:00   0h:00:06           -1638.9523       -       -\n",
      "  130/152  JTT-DCMUT+I    0h:00:00   0h:00:06           -1580.9748       -  0.2296\n",
      "  131/152  JTT-DCMUT+G4   0h:00:00   0h:00:06           -1558.1873  0.6529       -\n",
      "  132/152  JTT-DCMUT+I+G4 0h:00:00   0h:00:06           -1555.0198  1.4993  0.2021\n",
      "  133/152  JTT-DCMUT+F    0h:00:00   0h:00:06           -1649.3291       -       -\n",
      "  134/152  JTT-DCMUT+I+F  0h:00:00   0h:00:06           -1595.4239       -  0.2298\n",
      "  135/152  JTT-DCMUT+G4+F 0h:00:00   0h:00:06           -1571.3055  0.6515       -\n",
      "  136/152  JTT-DCMUT+I+G4+F0h:00:00   0h:00:06           -1568.5394  1.4428  0.1949\n",
      "  137/152  FLU            0h:00:00   0h:00:06           -1668.4819       -       -\n",
      "  138/152  FLU+I          0h:00:00   0h:00:06           -1611.4134       -  0.2258\n",
      "  139/152  FLU+G4         0h:00:00   0h:00:06           -1570.4725  0.5944       -\n",
      "  140/152  FLU+I+G4       0h:00:00   0h:00:06           -1568.1487  1.1025  0.1818\n",
      "  141/152  FLU+F          0h:00:00   0h:00:06           -1664.2143       -       -\n",
      "  142/152  FLU+I+F        0h:00:00   0h:00:06           -1616.2469       -  0.2250\n",
      "  143/152  FLU+G4+F       0h:00:01   0h:00:07           -1572.8255  0.5914       -\n",
      "  144/152  FLU+I+G4+F     0h:00:00   0h:00:07           -1571.2544  1.0990  0.1694\n",
      "  145/152  STMTREV        0h:00:00   0h:00:07           -1777.1868       -       -\n",
      "  146/152  STMTREV+I      0h:00:00   0h:00:07           -1695.1910       -  0.2299\n",
      "  147/152  STMTREV+G4     0h:00:00   0h:00:07           -1683.2497  0.5944       -\n",
      "  148/152  STMTREV+I+G4   0h:00:00   0h:00:07           -1676.2437  2.0599  0.2314\n",
      "  149/152  STMTREV+F      0h:00:00   0h:00:07           -1758.0823       -       -\n",
      "  150/152  STMTREV+I+F    0h:00:00   0h:00:07           -1682.5508       -  0.2300\n",
      "  151/152  STMTREV+G4+F   0h:00:00   0h:00:07           -1666.9461  0.6257       -\n",
      "  152/152  STMTREV+I+G4+F 0h:00:00   0h:00:07           -1661.1506  1.9089  0.2283\n",
      " ----ID---  ----MODEL---- ---Time--- -Elapsed--- -------LnL------- -Alpha- -P-inv-\n",
      "\n",
      "Computation of likelihood scores completed. It took 0h:00:07\n",
      "\n",
      "BIC       model              K            lnL          score          delta    weight\n",
      "--------------------------------------------------------------------------------\n",
      "       1  LG+G4              1     -1528.9082      3221.3709         0.0000    0.6335\n",
      "       2  LG+I+G4            2     -1527.3034      3222.4653         1.0944    0.3665\n",
      "       3  WAG+I+G4           2     -1539.1621      3246.1827        24.8118    0.0000\n",
      "       4  WAG+G4             1     -1541.6196      3246.7937        25.4229    0.0000\n",
      "       5  RTREV+G4           1     -1541.8393      3247.2330        25.8622    0.0000\n",
      "       6  RTREV+I+G4         2     -1540.4640      3248.7865        27.4156    0.0000\n",
      "       7  VT+I+G4            2     -1544.9913      3257.8412        36.4703    0.0000\n",
      "       8  VT+G4              1     -1547.6191      3258.7927        37.4218    0.0000\n",
      "       9  BLOSUM62+G4        1     -1549.0645      3261.6835        40.3127    0.0000\n",
      "      10  BLOSUM62+I+G4      2     -1547.4543      3262.7672        41.3963    0.0000\n",
      "--------------------------------------------------------------------------------\n",
      "Best model according to BIC\n",
      "---------------------------\n",
      "Model:              LG+G4\n",
      "lnL:                -1528.9082\n",
      "Frequencies:        0.0791 0.0559 0.0420 0.0531 0.0129 0.0408 0.0716 0.0573 0.0224 0.0622 0.0991 0.0646 0.0230 0.0423 0.0440 0.0612 0.0533 0.0121 0.0342 0.0691\n",
      "Inv. sites prop:    -\n",
      "Gamma shape:        0.6782\n",
      "Score:              3221.3709\n",
      "Weight:             0.6335\n",
      "---------------------------\n",
      "Parameter importances\n",
      "---------------------------\n",
      "P.Inv:              0.0000\n",
      "Gamma:              0.6335\n",
      "Gamma-Inv:          0.3665\n",
      "Frequencies:        0.0000\n",
      "---------------------------\n",
      "Model averaged estimates\n",
      "---------------------------\n",
      "P.Inv:              0.2252\n",
      "Alpha:              0.6782\n",
      "Alpha-P.Inv:        1.1341\n",
      "P.Inv-Alpha:        0.1633\n",
      "Frequencies:        0.1182 0.0608 0.0345 0.0534 0.0142 0.0162 0.0507 0.0595 0.0439 0.0547 0.0588 0.0250 0.0676 0.0392 0.0797 0.0466 0.0804 0.0108 0.0223 0.0635 \n",
      "\n",
      "Commands:\n",
      "  > phyml  -i Flavi_DEAD_2049_Gblocks.fasta -d aa -m LG -f m -v 0 -a e -c 4 -o tlr\n",
      "  > raxmlHPC-SSE3 -s Flavi_DEAD_2049_Gblocks.fasta -m PROTGAMMALG -n EXEC_NAME -p PARSIMONY_SEED\n",
      "  > raxml-ng --msa Flavi_DEAD_2049_Gblocks.fasta --model LG+G4\n",
      "  > paup -s Flavi_DEAD_2049_Gblocks.fasta\n",
      "  > iqtree -s Flavi_DEAD_2049_Gblocks.fasta -m LG+G4\n",
      "\n",
      "AIC       model              K            lnL          score          delta    weight\n",
      "--------------------------------------------------------------------------------\n",
      "       1  LG+I+G4            2     -1527.3034      3132.6067         0.0000    0.6468\n",
      "       2  LG+G4              1     -1528.9082      3133.8164         1.2096    0.3532\n",
      "       3  WAG+I+G4           2     -1539.1621      3156.3241        23.7174    0.0000\n",
      "       4  RTREV+I+G4         2     -1540.4640      3158.9279        26.3212    0.0000\n",
      "       5  WAG+G4             1     -1541.6196      3159.2393        26.6325    0.0000\n",
      "       6  RTREV+G4           1     -1541.8393      3159.6785        27.0718    0.0000\n",
      "       7  VT+I+G4            2     -1544.9913      3167.9826        35.3759    0.0000\n",
      "       8  VT+G4              1     -1547.6191      3171.2382        38.6315    0.0000\n",
      "       9  BLOSUM62+I+G4      2     -1547.4543      3172.9086        40.3019    0.0000\n",
      "      10  BLOSUM62+G4        1     -1549.0645      3174.1290        41.5223    0.0000\n",
      "--------------------------------------------------------------------------------\n",
      "Best model according to AIC\n",
      "---------------------------\n",
      "Model:              LG+I+G4\n",
      "lnL:                -1527.3034\n",
      "Frequencies:        0.0791 0.0559 0.0420 0.0531 0.0129 0.0408 0.0716 0.0573 0.0224 0.0622 0.0991 0.0646 0.0230 0.0423 0.0440 0.0612 0.0533 0.0121 0.0342 0.0691\n",
      "Inv. sites prop:    0.1633\n",
      "Gamma shape:        1.1341\n",
      "Score:              3132.6067\n",
      "Weight:             0.6468\n",
      "---------------------------\n",
      "Parameter importances\n",
      "---------------------------\n",
      "P.Inv:              0.0000\n",
      "Gamma:              0.3532\n",
      "Gamma-Inv:          0.6468\n",
      "Frequencies:        0.0000\n",
      "---------------------------\n",
      "Model averaged estimates\n",
      "---------------------------\n",
      "P.Inv:              0.2252\n",
      "Alpha:              0.6782\n",
      "Alpha-P.Inv:        1.1341\n",
      "P.Inv-Alpha:        0.1633\n",
      "Frequencies:        0.1182 0.0608 0.0345 0.0534 0.0142 0.0162 0.0507 0.0595 0.0439 0.0547 0.0588 0.0250 0.0676 0.0392 0.0797 0.0466 0.0804 0.0108 0.0223 0.0635 \n",
      "\n",
      "Commands:\n",
      "  > phyml  -i Flavi_DEAD_2049_Gblocks.fasta -d aa -m LG -f m -v e -a e -c 4 -o tlr\n",
      "  > raxmlHPC-SSE3 -s Flavi_DEAD_2049_Gblocks.fasta -m PROTGAMMAILG -n EXEC_NAME -p PARSIMONY_SEED\n",
      "  > raxml-ng --msa Flavi_DEAD_2049_Gblocks.fasta --model LG+I+G4\n",
      "  > paup -s Flavi_DEAD_2049_Gblocks.fasta\n",
      "  > iqtree -s Flavi_DEAD_2049_Gblocks.fasta -m LG+I+G4\n",
      "\n",
      "AICc      model              K            lnL          score          delta    weight\n",
      "--------------------------------------------------------------------------------\n",
      "       1  LG+G4              1     -1528.9082      3217.8164         0.0000    0.9476\n",
      "       2  LG+I+G4            2     -1527.3034      3223.6067         5.7904    0.0524\n",
      "       3  WAG+G4             1     -1541.6196      3243.2393        25.4229    0.0000\n",
      "       4  RTREV+G4           1     -1541.8393      3243.6785        25.8622    0.0000\n",
      "       5  WAG+I+G4           2     -1539.1621      3247.3241        29.5077    0.0000\n",
      "       6  RTREV+I+G4         2     -1540.4640      3249.9279        32.1115    0.0000\n",
      "       7  VT+G4              1     -1547.6191      3255.2382        37.4218    0.0000\n",
      "       8  BLOSUM62+G4        1     -1549.0645      3258.1290        40.3127    0.0000\n",
      "       9  VT+I+G4            2     -1544.9913      3258.9826        41.1663    0.0000\n",
      "      10  PMB+G4             1     -1550.6708      3261.3417        43.5253    0.0000\n",
      "--------------------------------------------------------------------------------\n",
      "Best model according to AICc\n",
      "---------------------------\n",
      "Model:              LG+G4\n",
      "lnL:                -1528.9082\n",
      "Frequencies:        0.0791 0.0559 0.0420 0.0531 0.0129 0.0408 0.0716 0.0573 0.0224 0.0622 0.0991 0.0646 0.0230 0.0423 0.0440 0.0612 0.0533 0.0121 0.0342 0.0691\n",
      "Inv. sites prop:    -\n",
      "Gamma shape:        0.6782\n",
      "Score:              3217.8164\n",
      "Weight:             0.9476\n",
      "---------------------------\n",
      "Parameter importances\n",
      "---------------------------\n",
      "P.Inv:              0.0000\n",
      "Gamma:              0.9476\n",
      "Gamma-Inv:          0.0524\n",
      "Frequencies:        0.0000\n",
      "---------------------------\n",
      "Model averaged estimates\n",
      "---------------------------\n",
      "P.Inv:              0.2252\n",
      "Alpha:              0.6782\n",
      "Alpha-P.Inv:        1.1341\n",
      "P.Inv-Alpha:        0.1633\n",
      "Frequencies:        0.1182 0.0608 0.0345 0.0534 0.0142 0.0162 0.0507 0.0595 0.0439 0.0547 0.0588 0.0250 0.0676 0.0392 0.0797 0.0466 0.0804 0.0108 0.0223 0.0635 \n",
      "\n",
      "Commands:\n",
      "  > phyml  -i Flavi_DEAD_2049_Gblocks.fasta -d aa -m LG -f m -v 0 -a e -c 4 -o tlr\n",
      "  > raxmlHPC-SSE3 -s Flavi_DEAD_2049_Gblocks.fasta -m PROTGAMMALG -n EXEC_NAME -p PARSIMONY_SEED\n",
      "  > raxml-ng --msa Flavi_DEAD_2049_Gblocks.fasta --model LG+G4\n",
      "  > paup -s Flavi_DEAD_2049_Gblocks.fasta\n",
      "  > iqtree -s Flavi_DEAD_2049_Gblocks.fasta -m LG+G4\n",
      "Summary:\n",
      "\n",
      "Partition 1/1:\n",
      "                         Model         Score        Weight\n",
      "----------------------------------------------------------\n",
      "       BIC               LG+G4     3221.3709        0.6335\n",
      "       AIC             LG+I+G4     3132.6067        0.6468\n",
      "      AICc               LG+G4     3217.8164        0.9476\n",
      "\n",
      "Execution results written to Flavi_DEAD_2049_Gblocks.fasta.out\n",
      "Starting tree written to Flavi_DEAD_2049_Gblocks.fasta.tree\n"
     ]
    }
   ],
   "source": [
    "# Modeltest\n",
    "!./modeltest-ng -i Flavi_DEAD_2049_Gblocks.fasta -d aa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1211fe8c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                 ..........................                                      \n",
      " ooooooooooooooooooooooooooooo        CURRENT SETTINGS        ooooooooooooooooooooooooooooooooooo\n",
      "                                 ..........................                                      \n",
      "\n",
      "                . Sequence filename:\t\t\t\t Flavi_DEAD_2049_Gblocks.phy\n",
      "                . Data type:\t\t\t\t\t aa\n",
      "                . Alphabet size:\t\t\t\t 20\n",
      "                . Sequence format:\t\t\t\t interleaved\n",
      "                . Number of data sets:\t\t\t\t 1\n",
      "                . Nb of bootstrapped data sets:\t\t\t 100\n",
      "                . Compute approximate likelihood ratio test:\t no\n",
      "                . Model name:\t\t\t\t\t LG\n",
      "                . Proportion of invariable sites:\t\t 0.000000\n",
      "                . Number of subst. rate categs:\t\t\t 4\n",
      "                . Gamma distribution parameter:\t\t\t estimated\n",
      "                . 'Middle' of each rate class:\t\t\t mean\n",
      "                . Amino acid equilibrium frequencies:\t\t model\n",
      "                . Optimise tree topology:\t\t\t yes\n",
      "                . Tree topology search:\t\t\t\t NNIs\n",
      "                . Starting tree:\t\t\t\t BioNJ\n",
      "                . Add random input tree:\t\t\t no\n",
      "                . Optimise branch lengths:\t\t\t yes\n",
      "                . Optimise substitution model parameters:\t yes\n",
      "                . Run ID:\t\t\t\t\t none\n",
      "                . Random seed:\t\t\t\t\t 1645046473\n",
      "                . Subtree patterns aliasing:\t\t\t no\n",
      "                . Version:\t\t\t\t\t 20120412\n",
      "\n",
      " oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo\n",
      "\n",
      "\n",
      "\n",
      ". 66 patterns found (out of a total of 74 sites). \n",
      "\n",
      ". 18 sites without polymorphism (24.32%).\n",
      "\n",
      ". Computing pairwise distances...\n",
      "\n",
      ". Building BioNJ tree...\n",
      "\n",
      ". This analysis requires at least 3 Mo of memory space.\n",
      "\n",
      ". Refining the input tree...\n",
      "\n",
      ". Maximizing likelihood (using NNI moves)...\n",
      "\n",
      ". (    2 sec) [     -1519.6318] [Alpha              ][  0.671883]\n",
      ". (    2 sec) [     -1519.6318] [Topology           ][# nnis=  1]\n",
      ". (    2 sec) [     -1519.0033] [Topology           ][# nnis=  0]\n",
      ". (    2 sec) [     -1518.4814] [Topology           ][# nnis=  0]\n",
      ". (    2 sec) [     -1518.3615] [Topology           ]\n",
      ". (    2 sec) [     -1518.3607] [Alpha              ][  0.676717]\n",
      ". (    2 sec) [     -1518.3607] [Topology           ][# nnis=  0]\n",
      ". (    3 sec) [     -1518.2972] [Topology           ][# nnis=  0]\n",
      ". (    3 sec) [     -1518.2560] [Topology           ]\n",
      ". (    3 sec) [     -1518.2560] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.2085] [Alpha              ][  0.676037]\n",
      ". (    3 sec) [     -1518.2085] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1706] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1706] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1535] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1535] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1448] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1448] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1395] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1395] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1361] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1361] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1340] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1340] [Branch lengths     ]\n",
      ". (    3 sec) [     -1518.1326] [Alpha              ][  0.676713]\n",
      ". (    3 sec) [     -1518.1326] [Branch lengths     ]\n",
      ". (    4 sec) [     -1518.1317] [Alpha              ][  0.676713]\n",
      ". (    4 sec) [     -1518.1317] [Alpha              ][  0.676713]\n",
      "\n",
      ". Checking for NNIs, optimizing five branches...\n",
      "\n",
      ". (    5 sec) [     -1517.7705] [Topology           ]\n",
      ". (    5 sec) [     -1517.7705] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7322] [Alpha              ][  0.694562]\n",
      ". (    5 sec) [     -1517.7322] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7202] [Alpha              ][  0.695678]\n",
      ". (    5 sec) [     -1517.7202] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7142] [Alpha              ][  0.695678]\n",
      ". (    5 sec) [     -1517.7142] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7106] [Alpha              ][  0.695678]\n",
      ". (    5 sec) [     -1517.7106] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7087] [Alpha              ][  0.695678]\n",
      ". (    5 sec) [     -1517.7087] [Branch lengths     ]\n",
      ". (    5 sec) [     -1517.7075] [Alpha              ][  0.695678]\n",
      ". (    5 sec) [     -1517.7075] [Branch lengths     ]\n",
      ". (    6 sec) [     -1517.7069] [Alpha              ][  0.695678]\n",
      ". (    6 sec) [     -1517.7069] [Alpha              ][  0.695678]\n",
      "\n",
      ". Checking for NNIs, optimizing five branches...\n",
      "\n",
      ". (    7 sec) [     -1517.7064] [Topology           ]\n",
      "\n",
      "\n",
      ". Log likelihood of the current tree: -1517.706355.\n",
      "\n",
      ". Launch bootstrap analysis on the most likely tree...\n",
      "\n",
      "\n",
      ". Non parametric bootstrap analysis \n",
      "\n",
      "  [....................]   20/ 100\n",
      "  [....................]   40/ 100\n",
      "  [....................]   60/ 100\n",
      "  [....................]   80/ 100\n",
      "  [....................]  100/ 100\n",
      "  \n",
      ". Printing the most likely tree in file 'Flavi_DEAD_2049_Gblocks.phy_phyml_tree.txt'...\n",
      "\n",
      ". Time used 0h6m49s\n",
      "\n",
      "oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo\n"
     ]
    }
   ],
   "source": [
    "# PhyML\n",
    "!phyml -i Flavi_DEAD_2049_Gblocks.phy -d aa -m LG -f m -a e -c 4 -v e -o tlr -b 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94541176",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
